src/System.Text.Json/src/System/Text/Json/Serialization/ReadStack.cs

```diff
@@ -53,15 +53,21 @@ public string PropertyPath
         {
             get
             {
-                StringBuilder path = new StringBuilder();
+                StringBuilder path;
 
                 if (_previous == null || _index == 0)
                 {
-                    path.Append($"[{Current.JsonClassInfo.Type.FullName}]");
+                    // No path if we've walked beyond the end of our JSON document
+                    if (Current.JsonClassInfo == null)
+                    {
+                        return "<none>";
+                    }
+
+                    path = new StringBuilder($"[{Current.JsonClassInfo.Type.FullName}]");
                 }
                 else
                 {
-                    path.Append($"[{_previous[0].JsonClassInfo.Type.FullName}]");
+                    path = new StringBuilder($"[{_previous[0].JsonClassInfo.Type.FullName}]");
```

Comment:
Can `_previous[0].JsonClassInfo` ever be null, similar to the `if (Current.JsonClassInfo == null)` case above?

Reply:
I tried various levels of invalid nesting or unexpected types and could not elicit an NPE down this path.

Reply:
Great. Good to know :)